Add delete user functionality to manage.py

redshiftzero opened the pull request:
Implements #776, enabling an admin to quickly delete a user from the application using `manage.py`:

```
vagrant@development:/vagrant/securedrop$ ./manage.py delete-user
Username to delete: not-a-user
ERROR: That user was not found!
Username to delete: redshiftzero
User 'redshiftzero' successfully deleted
vagrant@development:/vagrant/securedrop$ 
```

fowlslegs commented:
I kind of agree with Garrett's comment https://github.com/freedomofpress/securedrop/issues/776#issuecomment-65827651 in that I don't see a real purpose for this. It seems like adding a feature just for the sake of it.

redshiftzero commented:
ok will close my PR and the underlying issue

redshiftzero commented:
Ok after reading #344, #225, and given the discussion at the developer meeting about how ideally the common administrator functions should be possible through a CLI (and then one day through the web interface and/or a GUI), I'm gonna re-open this one. See in particular [this comment](https://github.com/freedomofpress/securedrop/issues/225#issuecomment-59860216).

redshiftzero commented:
Sure! I removed the second `try except` block (I agree with the badness of `except Exception`) and rebased on current develop 😸 . The exception handling in general in this file could use some additional work, but that's a PR for another day.

fowlslegs commented:
I used my admin powers to merge this despite coveralls failing due to a 0.5% decrease in code coverage. The reason I did not ask Jen to write tests for this is that I'm in the middle of a major rewrite of this component that should land early next week. Further, the code is very minimal, mostly for testing purposes, and near identical to what we do in `journalist.py`.
